HealthcareSystem/GraphicalEditor/MainWindow.xaml.cs

```diff
@@ -126,33 +127,84 @@ private void saveMap()
+                SaveButton.Visibility = Visibility.Hidden;
+                EditMode = false ;
+            }
+            else
+            {
+                if (_selectedMapObject.MapObjectEntity.MapObjectType.TypeOfMapObject != TypeOfMapObject.ROAD)
+                {
+                    EditObjectButton.Visibility = Visibility.Visible;
+                    SaveButton.Visibility = Visibility.Visible;
+                }
+                else {
+                    EditObjectButton.Visibility = Visibility.Hidden;
+                    SaveButton.Visibility = Visibility.Hidden;
+                }
+            }
+        }
+
+        private MapObjectType SettingTypeOfEditedMapObject() {
+            MapObjectType type;
+            if (ObjectTypeChooserComboBox.SelectedItem != null)
+            {
+                type = (MapObjectType)ObjectTypeChooserComboBox.SelectedItem;
+            }
+            else
+            {
+                type = SelectedMapObject.MapObjectEntity.MapObjectType;
+            }
+            return type;
+        }
+
+        private MapObject SettingObjectForEdition(MapObjectType type) {
             MapObject objectToEdit = SelectedMapObject;
+            DisplayMapObject.MapObjectType = type;
             objectToEdit.MapObjectEntity = DisplayMapObject;
-            MapObjectType type = new MapObjectType();
-            type.ObjectTypeFullName = DisplayMapObject.MapObjectType.ObjectTypeFullName;           
-            MapObjectEntity newEntity = new MapObjectEntity(type.TypeOfMapObject, DisplayMapObject.Description);
-            MapObject newObject = new MapObject(newEntity, SelectedMapObject.MapObjectMetrics, SelectedMapObject.MapObjectDoor);
-            newObject.MapObjectEntity.Id = SelectedMapObject.MapObjectEntity.Id;
-            _mapObjectController.UpdateMapObject(objectToEdit);                      
+            objectToEdit.Rectangle.Fill = type.ObjectTypeColor;
+            return objectToEdit;
+        }
+
+        private MapObject CreateEditedObject() {
+            MapObjectType type = SettingTypeOfEditedMapObject();
+            MapObject objectToEdit = SettingObjectForEdition(type);
+            _mapObjectController.UpdateMapObject(objectToEdit);
+            return objectToEdit;
+        }
+
+        private void ReloadMap(MapObject objectToEdit) {
             EditMode = !EditMode;
+            Canvas.Children.Clear();
+            SelectedMapObject.MapObjectEntity.MapObjectType = objectToEdit.MapObjectEntity.MapObjectType;
+            LoadMapOnCanvas();
         }
 
-        private void Cancel_Editing_Mode(object sender, RoutedEventArgs e)
+        private void Change_Display_Information(object sender, RoutedEventArgs e)
         {
+            ReloadMap(CreateEditedObject());
+        }
+
+        private void Cancel_Editing_Mode(object sender, RoutedEventArgs e)
+        {            
+            Canvas.Children.Clear();
+            LoadMapOnCanvas();
```

Comment:
Why is it necessary to reload the map in order to cancel editing?

Reply:
so all object on map are shown properly immediately